packages/parser/src/language/pie/pieTokenBuilder.ts

```diff
@@ -0,0 +1,23 @@
+import type { GrammarAST, Stream, TokenBuilderOptions } from 'langium';
+import { DefaultTokenBuilder } from 'langium';
+
+import type { TokenType } from '../chevrotainWrapper.js';
+
+export class PieTokenBuilder extends DefaultTokenBuilder {
+  protected override buildKeywordTokens(
+    rules: Stream<GrammarAST.AbstractRule>,
+    terminalTokens: TokenType[],
+    options?: TokenBuilderOptions
+  ): TokenType[] {
+    const tokenTypes: TokenType[] = super.buildKeywordTokens(rules, terminalTokens, options);
+    tokenTypes.forEach((tokenType: TokenType): void => {
+      if (
+        (tokenType.name === 'pie' || tokenType.name === 'showData') &&
+        tokenType.PATTERN !== undefined
+      ) {
+        tokenType.PATTERN = new RegExp(tokenType.PATTERN.toString() + '(?!\\S)');
+      }
+    });
```

Comment:
See https://github.com/mermaid-js/mermaid/commit/11b60ce3b861374ed571a52f2d34ad8cf6e611cc

Reply:
I wouldn't recommend name the abstract class as `MermaidTokenBuilder`, it would confuse contributors when implementing new parser.
We mostly extend `Common...` classes in other diagram services, but when it comes to `TokenBuilder`, extending`MermaidTokenBuilder` is kinda weird.

Reply:
MermaidTokenBuilder -> The abstract class 
CommonTokenBuilder -> Implementation of MermaidTokenBuilder that is used only in places where a specific TokenBuilder is not necessary
<Diagram>TokenBuilder -> Implementation of MermaidTokenBuilder for that particular diagram.

Reply:
Done 69198ccccd8d4082461d3fbd45d43de8bb05f70e